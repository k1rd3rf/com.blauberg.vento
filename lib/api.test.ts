import Api from '../drivers/vento-expert/driver';
import { statusResponse } from './__mockdata__/statusResponse';
import { removeUndefinedDeep } from './testTools';
import { sendMock } from './__mocks__/blaubergventojs';

describe('Api set functions', () => {
  let api: Api;
  afterEach(() => {
    jest.clearAllMocks();
  });

  beforeEach(async () => {
    jest.spyOn(global.console, 'error');
    api = new Api();
    await api.onInit();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
  beforeEach(() => {
    sendMock.mockImplementation((packet, ip) =>
      Promise.resolve({ packet, ip })
    );
  });

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const getMockCalls = (response: any) => ({
    calls: sendMock.mock.calls,
    response: removeUndefinedDeep(response),
  });
  const device = { id: 'fanId', ip: '127.0.0.1' };

  it('update sends correct data', async () => {
    const response = await api.getDeviceState(device, 'password');
    expect(getMockCalls(response)).toMatchSnapshot();
  });
  it('update gets response', async () => {
    sendMock.mockResolvedValue(statusResponse);
    const response = await api.getDeviceState(device, 'password');
    expect(getMockCalls(response)).toMatchSnapshot();
  });

  [0, 1, 2].forEach((value) => {
    it(`setOnOffStatus sends value ${value}`, async () => {
      const response = await api.setOnOffStatus(device, 'password', value);
      expect(getMockCalls(response)).toMatchSnapshot();
    });
    it(`setSpeedMode sends value ${value}`, async () => {
      const response = await api.setSpeedMode(device, 'password', value);
      expect(getMockCalls(response)).toMatchSnapshot();
    });
    it(`setOperationMode sends value ${value}`, async () => {
      const response = await api.setOperationMode(device, 'password', value);
      expect(getMockCalls(response)).toMatchSnapshot();
    });
    it(`setTimerMode sends value ${value}`, async () => {
      const response = await api.setTimerMode(device, 'password', value);
      expect(getMockCalls(response)).toMatchSnapshot();
    });
    it(`setManualSpeed sends value ${value}`, async () => {
      const response = await api.setManualSpeed(device, 'password', value);
      expect(getMockCalls(response)).toMatchSnapshot();
    });
    it(`setHumiditySensor sends value ${value}`, async () => {
      const response = await api.setHumiditySensor(device, 'password', value);
      expect(getMockCalls(response)).toMatchSnapshot();
    });
    it(`setHumiditySensorThreshold sends value ${value}`, async () => {
      const response = await api.setHumiditySensorThreshold(
        device,
        'password',
        value
      );
      expect(getMockCalls(response)).toMatchSnapshot();
    });
    it(`setBoostDelay sends value ${value}`, async () => {
      const response = await api.setBoostDelay(device, 'password', value);
      expect(getMockCalls(response)).toMatchSnapshot();
    });
  });
});
