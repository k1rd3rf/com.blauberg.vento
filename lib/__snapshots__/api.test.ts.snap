// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Api set functions setBoostDelay sends value 0 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 102,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": 0,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setBoostDelay sends value 1 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 102,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": 1,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setBoostDelay sends value 2 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 102,
            "value": Uint8Array [
              2,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": 2,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setHumiditySensor sends value 0 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 15,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": 0,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setHumiditySensor sends value 1 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 15,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": 1,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setHumiditySensor sends value 2 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 15,
            "value": Uint8Array [
              2,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": 2,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setHumiditySensorThreshold sends value 0 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 25,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": 0,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setHumiditySensorThreshold sends value 1 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 25,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": 1,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setHumiditySensorThreshold sends value 2 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 25,
            "value": Uint8Array [
              2,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": 2,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setManualSpeed sends value 0 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 68,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": 0,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setManualSpeed sends value 1 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 68,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": 1,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setManualSpeed sends value 2 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 68,
            "value": Uint8Array [
              2,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": 2,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setOnOffStatus sends value 0 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 1,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": 0,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setOnOffStatus sends value 1 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 1,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": 1,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setOnOffStatus sends value 2 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 1,
            "value": Uint8Array [
              2,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": 2,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setOperationMode sends value 0 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 183,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": 0,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setOperationMode sends value 1 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 183,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": 1,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setOperationMode sends value 2 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 183,
            "value": Uint8Array [
              2,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": 2,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setSpeedMode sends value 0 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 2,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": 0,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setSpeedMode sends value 1 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 2,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": 1,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setSpeedMode sends value 2 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 2,
            "value": Uint8Array [
              2,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": 2,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setTimerMode sends value 0 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 7,
            "value": Uint8Array [
              0,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": 0,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setTimerMode sends value 1 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 7,
            "value": Uint8Array [
              1,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": 1,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions setTimerMode sends value 2 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 7,
            "value": Uint8Array [
              2,
            ],
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 2,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": 2,
    },
    "unittype": "Vento Expert",
  },
}
`;

exports[`Api set functions update gets response 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 1,
            "value": undefined,
          },
          {
            "parameter": 2,
            "value": undefined,
          },
          {
            "parameter": 68,
            "value": undefined,
          },
          {
            "parameter": 6,
            "value": undefined,
          },
          {
            "parameter": 102,
            "value": undefined,
          },
          {
            "parameter": 183,
            "value": undefined,
          },
          {
            "parameter": 136,
            "value": undefined,
          },
          {
            "parameter": 100,
            "value": undefined,
          },
          {
            "parameter": 37,
            "value": undefined,
          },
          {
            "parameter": 15,
            "value": undefined,
          },
          {
            "parameter": 25,
            "value": undefined,
          },
          {
            "parameter": 185,
            "value": undefined,
          },
          {
            "parameter": 74,
            "value": undefined,
          },
          {
            "parameter": 7,
            "value": undefined,
          },
          {
            "parameter": 131,
            "value": undefined,
          },
          {
            "parameter": 11,
            "value": undefined,
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 1,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": 0,
    "boost": {
      "deactivationtimer": 30,
      "mode": 0,
    },
    "fan": {
      "rpm": 236,
    },
    "filter": {
      "alarm": 0,
      "timer": {
        "days": 31,
        "hour": 20,
        "min": 34,
      },
    },
    "humidity": {
      "activated": 0,
      "current": 45,
      "sensoractivation": 1,
      "threshold": 70,
    },
    "onoff": 1,
    "operationmode": 2,
    "speed": {
      "manualspeed": 179,
      "mode": 1,
    },
    "timers": {
      "countdown": {
        "hour": 0,
        "min": 0,
        "sec": 0,
      },
      "mode": 0,
    },
    "unittype": "Vento Expert Duo A30-1 W V.2",
  },
}
`;

exports[`Api set functions update sends correct data 1`] = `
{
  "calls": [
    [
      E {
        "_dataEntries": [
          {
            "parameter": 1,
            "value": undefined,
          },
          {
            "parameter": 2,
            "value": undefined,
          },
          {
            "parameter": 68,
            "value": undefined,
          },
          {
            "parameter": 6,
            "value": undefined,
          },
          {
            "parameter": 102,
            "value": undefined,
          },
          {
            "parameter": 183,
            "value": undefined,
          },
          {
            "parameter": 136,
            "value": undefined,
          },
          {
            "parameter": 100,
            "value": undefined,
          },
          {
            "parameter": 37,
            "value": undefined,
          },
          {
            "parameter": 15,
            "value": undefined,
          },
          {
            "parameter": 25,
            "value": undefined,
          },
          {
            "parameter": 185,
            "value": undefined,
          },
          {
            "parameter": 74,
            "value": undefined,
          },
          {
            "parameter": 7,
            "value": undefined,
          },
          {
            "parameter": 131,
            "value": undefined,
          },
          {
            "parameter": 11,
            "value": undefined,
          },
        ],
        "_deviceId": "fanId",
        "_functionType": 1,
        "_password": "password",
      },
      "127.0.0.1",
    ],
  ],
  "response": {
    "alarm": undefined,
    "boost": {
      "deactivationtimer": undefined,
      "mode": undefined,
    },
    "fan": {
      "rpm": undefined,
    },
    "filter": {
      "alarm": undefined,
      "timer": {
        "days": undefined,
        "hour": undefined,
        "min": undefined,
      },
    },
    "humidity": {
      "activated": 0,
      "current": undefined,
      "sensoractivation": undefined,
      "threshold": undefined,
    },
    "onoff": undefined,
    "operationmode": undefined,
    "speed": {
      "manualspeed": undefined,
      "mode": undefined,
    },
    "timers": {
      "countdown": {
        "hour": undefined,
        "min": undefined,
        "sec": undefined,
      },
      "mode": undefined,
    },
    "unittype": "Vento Expert",
  },
}
`;
